### PR TITLE
Added experiment_id to NeptuneLogger

### PR DESCRIPTION
**Pull request by blatr**
PR for #3389.
Probably need to add a test case for the new behaviour but i'd rather wait #3256 to finalize their test cases format.

### CODE REVIEW

**Comment by Borda** [comment]
@blatr how is it going, is it still WIP?

**Comment by blatr** [comment]
> @blatr how is it going, is it still WIP?

Actually I've made all the changes for my proposal except the test case. 
As for test case I can't write it right now because @pitercl proposed a new logic behind experiment initialization in [3256](https://github.com/PyTorchLightning/pytorch-lightning/pull/3256) and I'm a little confused about whether I should write test case based on old logic or wait until that issue will be resolved and use proposed new logic.

**Comment by Borda** [comment]
@blatr @pitercl any update here?

**Comment by pitercl** [comment]
For me this PR looks good (aside maybe from what @blatr mentioned - a missing additional test case).

What I proposed in #3256 is only a way to rewrite tests that are failing after changes - nothing major. I think we can safely merge this PR (after adding a test case :smile: ). An additional test case here could look like this:

```py3
@patch('pytorch_lightning.loggers.neptune.neptune')
def test_neptune_existing_experiment(neptune):
    logger = NeptuneLogger(experiment_id='TEST-123')

    neptune.Session.with_default_backend().get_project().get_experiments.assert_called_once_with(id='TEST-123')

    experiment = logger.experiment

    assert logger.properties == experiment.get_properties()
    # and tags, params
 ```

**Comment by blatr** [comment]
Thanks @pitercl. I've added proposed test case to PR. Now everything is done.
However after merging with current "master" branch some CI checks have become unsuccesful.  
And now I'm a bit stuck on understanding what's exactly broken. @Borda @pitercl Can you help me please to understand what's wrong now and what should i fix in order to pass all tests? Sorry to bother you, It's my first issue and i'm not feeling confident about it.

**Comment by pitercl** [comment]
First of all, thank you for contributing :) It's truly awesome to see someone care enough to do that :)

As for the PR, one thing that got broken after the merge from master is [this line](https://github.com/PyTorchLightning/pytorch-lightning/pull/3462/files#diff-4bc801846ba98d3e6d6a8e7dff3f49bcR371). It should be `self.experiment_id`, as in your original commit. I haven't tested it, but I'd expect tests to pass after this fix.

Another, separate topic is that the changes merged from master moved some arguments like `tags`, `properties`, etc to `**kwargs`. For my taste `experiment_id` is prominent enough to leave it explicitly in the `__init__`'s argument list, but I'll leave the final judgement to @Borda.

**Comment by pitercl** [comment]
The tests look good now! :+1:  Unfortunately, the remaining issue is not something I know how to help with. Maybe @Borda?

**Comment by Borda** [comment]
@pitercl you can ignore Docker checks now as it shall be fixed in #3765

**Comment by blatr** [comment]
@pitercl thx again:) 
@Borda yep!

File: pytorch_lightning/loggers/neptune.py
Comment (Borda) [inline pytorch_lightning/loggers/neptune.py:380]: why this assignment if the `self.experiment_id` is not empty?

Comment (blatr) [reply pytorch_lightning/loggers/neptune.py:380]: I suppose that's the only way ```self.experiment_id``` won't be empty inside ```_create_or_get_experiment()``` is passing it as an argument during a class initialization.  
But if it's passed as an argument ```exp = project.get_experiments(id=self.experiment_id)[0]``` will retrieve full information about the experiment with this experiment_id and ```exp.id``` will be equal to ```self.experiment_id``` as a result.  
Actually this assigment is needed only when new experiment was created. So, we can rearrange code like this:
```python
    def _create_or_get_experiment(self):
        if self.offline_mode:
            project = neptune.Session(backend=neptune.OfflineBackend()).get_project('dry-run/project')
        else:
            session = neptune.Session.with_default_backend(api_token=self.api_key)
            project = session.get_project(self.project_name)

        if self.experiment_id is None:
            exp = project.create_experiment(name=self.experiment_name, **self._kwargs)
            self.experiment_id = exp.id #<-------move assigment here 
        else:
            exp = project.get_experiments(id=self.experiment_id)[0]
            self.experiment_name = exp.get_system_properties()['name']
            self.params = exp.get_parameters()
            self.properties = exp.get_properties()
            self.tags = exp.get_tags()

        return exp```

**Review by awaelchli — APPROVED** [review]

**Comment by 0560405608** [comment]
@blatr ****

**Review by tchaton — APPROVED** [review]

File: pytorch_lightning/loggers/neptune.py
Comment (rohitgr7) [inline pytorch_lightning/loggers/neptune.py:167]:
```suggestion
        experiment_id: Optional. Default is ``None``. The ID of the existing experiment.
            If specified, connect to experiment with experiment_id in project_name.
            Input arguments "experiment_name", "params", "properties" and "tags" will be overriden based
            on fetched experiment data.
```

**Comment by rohitgr7** [comment]
A test seems to be failing on master. @awaelchli can you check this one on windows??

**Comment by awaelchli** [comment]
@rohitgr7 Fixed here #3821 by Jeff, we need to exclude ddp windows for the metrics for now

**Comment by rohitgr7** [comment]
ok great. I though it was something related to pathlib + windows

File: pytorch_lightning/loggers/neptune.py
Comment (rohitgr7) [inline pytorch_lightning/loggers/neptune.py:192]: experiment_id can't be private here?
```suggestion
        self._experiment_id = experiment_id
```

**Review by Borda — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]

**Pull request merged**
